icu4c/source/test/intltest/cplusplus_header_api_build_test.cpp

```diff
@@ -0,0 +1,11 @@
+// © 2025 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html#License
+
+// ICU-22954 Test that client code can be built with U_SHOW_CPLUSPLUS_API=0.
+#undef U_ALL_IMPLEMENTATION
+#undef U_SHOW_CPLUSPLUS_API
+#define U_SHOW_CPLUSPLUS_API 0
+#include "unicode/char16ptr.h"
+#include "unicode/ucol.h"
```

Comment:
This proves that ucol.h does not cause compiler errors when included in this configuration, but it doesn't prove that these predicates work like this. This file here would be just as happy if ucol.h looked empty...

Why not move the relevant unit tests into a file like https://github.com/unicode-org/icu/blob/main/icu4c/source/test/intltest/usetheaderonlytest.cpp ?

----
otherwise lgtm tnx!

Reply:
Done.